src/site/content/en/blog/bfcache/index.md

```diff
@@ -303,7 +303,15 @@ onAllChangesSaved(() => {
 {% endCompare %}
 
 
-### Avoid window.opener references
+### Do not serve pages with `Cache-control: no-store`
```

Comment:
```suggestion
### Avoid `Cache-Control: no-store`
```

Are there really no good reasons to use `no-store`? If we really do want to always discourage it then we should probably give some examples of how you can prevent surprising users in situations like if the user logs out and then clicks the back button (see other comments).

Reply:
I think ensuring that logged-in-only information is not re-presented is a legit use. pageshow handlers could do something close but reliably hiding sensitive information there is tricky (without just deleting the whole DOM) and maybe defeatable with devtools. Basically it's the simplest and most reliable method.

Reply:
Yeah that's why I had "should be used unless there are good security reasons for using `Cache-control: no-store`." below. Maybe should expand upon that?

However, Chrome is also planning on clearing the bfcache once cookies change before enabling the `no-store` use case (I presume that is still the case @fergald ?), which should cover this case too. So I'm kinda reluctant to recommend that's `no-store` is used for this case, then in the (near?) future we override that. Hence why I left it a little vague with that line.

Reply:
Our intention there is to stop over-blocking and that has to be done in a safe way, so anyone using CCNS to protect logged-in-only info is correct to do so and should continue doing so.

Reply:
Gotcha. So guidance is: `no-store` for sensitive information and `no-cache`/`max-age=0` for avoiding stale information without revalidated and we just think the former is being over used at present for reasons covered by the latter. And we think that often not revalidating on bfcache restore is actually OK.

In that case think we should make that clear. But will see where we land on the current comments before tackling that.

Reply:
It's not that it's being overused (well it also is but assuming it's actually being used appropriately) it's that it's too blunt an instrument. We never cache, just in case we cache after logout. We know that's 90% of the time, it would have been fine to cache. The proposal is for us to start figuring that out and to let pages tell us directly.

Reply:
Right but, since a web site can’t control this the advice is still `no-store` on sensitive information. For example, they are at /private-page and then navigate to /logout to prevent restoring on back they should use `no-store` on /private-page. At some point we (Chrome) might handle that better by allowing certain “backs” despite `no-store` but advice for developer wouldn’t change even then - use `no-store` and let us worry about that (with extra checks via JavaScript on restore if really necessary).

But pages without private info, but with “fresh” preference (e.g. news publisher) then `no-store` is overkill and should not be used as restoring back to that page is more likely what the user expects.

That about sum up the situation? Will think of how to express this is an easy to understand way for the post!

Reply:
Yes. Thanks.